src/trading/getPreSignTransaction.ts

```diff
@@ -17,18 +23,9 @@ export async function getPreSignTransaction(
   const settlementContractAddress = COW_PROTOCOL_SETTLEMENT_CONTRACT_ADDRESS[chainId] as `0x${string}`
   const preSignatureCall = contract.interface.encodeFunctionData('setPreSignature', [orderId, true])
 
-  const gas = await contract.estimateGas
-    .setPreSignature(orderId, true)
-    .then((res) => BigInt(res.toHexString()))
-    .catch((error) => {
-      console.error(error)
-
-      return GAS_LIMIT_DEFAULT
-    })
-
   return {
     data: preSignatureCall,
-    gas: '0x' + calculateGasMargin(gas).toString(16),
+    gas: '0x' + PRE_SIGN_GAS_LIMIT.toString(16),
```

Comment:
my concern is that this is not hardfork proof, its not the first time ethereum changes the gas costs for some operations

Also, it overestimates

Reply:
Yes, it overestimates in case of "ethereum changes the gas costs".
Since this is our smart-contract we can trust it and give a gasLimit with enough buffer.
I don't insist on this solution, the guy from Gnosis who was using this SDK for the hackhaton complained about it.
Anyway, this improvement might me skipped.

Reply:
In case when the node RPC doesn't support eth_estimateGas we just fallback to 150k gas limit

Reply:
Why would the RPC not support `eth_estimateGas`, this is what I don't get